pkg/plugins/optional/grafana/v1alpha/scaffolds/edit.go

```diff
@@ -41,15 +47,122 @@ func (s *editScaffolder) InjectFS(fs machinery.Filesystem) {
 	s.fs = fs
 }
 
+func fileExist(configFilePath string) bool {
+	if _, err := os.Stat(configFilePath); os.IsNotExist(err) {
+		return false
+	}
+	return true
+}
+
+func loadConfig() ([]templates.CustomMetricItem, error) {
+	// TODO: Is it good to hard code the config file path here?
+	configPath := "grafana/custom-metrics/config.yaml"
+
+	if !fileExist(configPath) {
+		return nil, nil
+	}
+
+	f, err := os.Open(configPath)
+	if err != nil {
+		return nil, fmt.Errorf("error loading plugin config: %w", err)
+	}
+
+	items, err := configReader(f)
+
+	if err := f.Close(); err != nil {
+		return nil, fmt.Errorf("could not close config.yaml: %w", err)
+	}
+
+	return items, err
+}
+
+func configReader(reader io.Reader) ([]templates.CustomMetricItem, error) {
+	yamlFile, err := ioutil.ReadAll(reader)
+	if err != nil {
+		return nil, err
+	}
+
+	config := templates.CustomMetricsConfig{}
+
+	err = yaml.Unmarshal(yamlFile, &config)
+	if err != nil {
+		return nil, err
+	}
+
+	validatedMetricItems := validateCustomMetricItems(config.CustomMetrics)
+
+	return validatedMetricItems, nil
+}
+
+func validateCustomMetricItems(rawItems []templates.CustomMetricItem) []templates.CustomMetricItem {
+	// 1. Filter items of missing `Metric` or `Type`
+	filterResult := []templates.CustomMetricItem{}
+	for _, item := range rawItems {
+		if fieldsExist(item) {
+			filterResult = append(filterResult, item)
+		}
+	}
+
+	// 2. Fill Expr if missing
+	validatedItems := make([]templates.CustomMetricItem, len(filterResult))
+	for i, item := range filterResult {
+		validatedItems[i] = fillMissingExpr(item)
+	}
+
+	return validatedItems
+}
+
+func fieldsExist(item templates.CustomMetricItem) bool {
+	// If `Expr` exists, return true
+	if item.Expr != "" {
+		return true
+	}
+
+	// If `Metric` & valid `Type` exists, return true
+	metricType := strings.ToLower(item.Type)
+	if item.Metric != "" && (metricType == "counter" || metricType == "gauge" || metricType == "histogram") {
+		return true
+	}
+
+	return false
+}
+
+func fillMissingExpr(item templates.CustomMetricItem) templates.CustomMetricItem {
```

Comment:
so this func adds a default expression when user does not provide it?
If so, is it compatible with all the three kinds of prometheus metrics like gauge, counter etc.